src/code_80850_len_3060.c

```diff
@@ -268,7 +268,108 @@ void enforce_hpfp_limits(void) {
     }
 }
 
-INCLUDE_ASM(s32, "code_80850_len_3060", initialize_status_menu);
+void initialize_status_menu() {
+    UiStatus* uiStatus = &gUIStatus;
+    PlayerData* playerData = &gPlayerData;
+    s32 var0;
+    s32 var1;
+    s32 new_var;
+
+    uiStatus->drawPosX = 0x0C;
+    D_8010CD10 = 0;
+    D_8010CD12 = 0;
+    uiStatus->drawPosY = -0x64;
+    uiStatus->hidden = 0;
+    uiStatus->showTimer = 0xD2;
+    uiStatus->unk_3B[0] = 0;
+    uiStatus->ignoreChanges = 0;
+    uiStatus->unk_45[0] = 0;
+    uiStatus->unk_45[1] = 0;
+    uiStatus->hpBlinking = 0;
+    uiStatus->hpBlinkCounter = 0;
+    uiStatus->hpBlinkTimer = 0;
+    uiStatus->fpBlinking = 0;
+    uiStatus->fpBlinkCounter = 0;
+    uiStatus->fpBlinkTimer = 0;
+    uiStatus->spBlinking = 0;
+    uiStatus->spBlinkCounter = 0;
+    uiStatus->coinsBlinking = 0;
+    uiStatus->coinsBlinkCounter = 0;
+    uiStatus->coinsBlinkTimer = 0;
+    uiStatus->disabled = 0;
+    uiStatus->starpointsBlinking = 0;
+    uiStatus->displayHP = playerData->curHP;
+    uiStatus->displayFP = playerData->curFP;
+    uiStatus->displayCoins = playerData->coins;
+    uiStatus->displayStarpoints = playerData->starPoints;
+    uiStatus->displaySP = playerData->specialBarsFilled;
+    uiStatus->starpointsBlinkCounter = 0;
+    uiStatus->unk_6C[2] = -1;
+    uiStatus->unk_3B[1] = 0;
+    uiStatus->unk_57[0] = 0;
+    uiStatus->unk_57[1] = 0;
+    uiStatus->unk_57[2] = 0;
+    uiStatus->unk_57[3] = 0;
+    uiStatus->unk_6C[0] = 0;
+    uiStatus->unk_6C[1] = 0;
+    uiStatus->iconIndex12 = -1;
+    func_800E97B8();
+    var0 = create_icon(&D_80108248);
+    uiStatus->hpIconIndexes[0] = var0;
+    set_icon_flags(var0, 0x80);
+    clear_icon_flags(var0, 0x8000);
+    var0 = create_icon(&D_80108518);
+    uiStatus->hpIconIndexes[1] = var0;
+    set_icon_flags(var0, 0x80);
+    clear_icon_flags(var0, 0x8000);
+    var0 = create_icon(&D_80108270);
+    uiStatus->fpIconIndexes[0] = var0;
+    set_icon_flags(var0, 0x80);
+    clear_icon_flags(var0, 0x8000);
+    var0 = create_icon(&D_80108538);
+    uiStatus->fpIconIndexes[1] = var0;
+    set_icon_flags(var0, 0x80);
+    clear_icon_flags(var0, 0x8000);
+    var0 = create_icon(&D_80108558);
+    uiStatus->coinIconIndex = var0;
+    set_icon_flags(var0, 0x80);
+    clear_icon_flags(var0, 0x8000);
+    var0 = create_icon(&D_80080FC4);
+    uiStatus->coinIconIndex2 = var0;
+    set_icon_flags(var0, 0x80);
+    clear_icon_flags(var0, 0x8000);
+    var0 = create_icon(&D_801086AC);
+    uiStatus->starpointsIconIndex = var0;
+    set_icon_flags(var0, 0x80);
+    clear_icon_flags(var0, 0x8000);
+    var0 = create_icon(&D_80104BEC);
+    uiStatus->starpointsIconIndex2 = var0;
+    set_icon_flags(var0, 0x80);
+    clear_icon_flags(var0, 0x8000);
+    var1 = &D_80108068;
+    new_var = create_icon(&D_80108068);
```

Comment:
are you sure new_var is needed here? should be able to just do var0 = like the rest of these ideally

Reply:
just doing
`iconIndex = create_icon(&D_80108068);`
makes s0 and s1 swap in the code